delivery/services/dds_service.py

```diff
@@ -158,6 +158,10 @@ def _run_dds_put(
             # Always commit the state change to the database
             session.commit()
 
+        log.info(
```

Comment:
To me it would make sense to put the cleanup in the `finally` block.

Reply:
Yes good point :+1: !